java/src/org/openqa/selenium/grid/distributor/config/DistributorOptions.java

```diff
@@ -75,14 +77,23 @@ public URI getDistributorUri() {
     }
 
     try {
-      return new URI("http", null, hostname.get(), port.get(), null, null, null);
+      return new URI(schema, null, hostname.get(), port.get(), null, null, null);
     } catch (URISyntaxException e) {
       throw new ConfigException(
           "Distributor uri configured through host (%s) and port (%d) is not a valid URI",
           hostname.get(), port.get());
     }
   }
 
+  public boolean isSecure() {
+    return config.get(SERVER_SECTION, "https-private-key").isPresent()
+        && config.get(SERVER_SECTION, "https-certificate").isPresent();
+  }
+
+  public boolean isSelfSigned() {
+    return config.getBool(SERVER_SECTION, "https-self-signed").orElse(false);
+  }
+
```

Comment:
This could be implemented in `BaseServerOptions` and reuse across other classes.